### ISSUE TITLE
DLSS related settings added

### ISSUE THREAD

**Issue by emoose** [Orbmu2k/nvidiaProfileInspector#156]
DLSS 3.1.11 started reading a few settings from DRS, here's what I found in 3.1.30:

```
0x10E41DF3 (NGX_OVERRIDE_RENDER_PRESET_SELECTION) - dword (0 - 6 which goes to preset A/B/C/D etc)
0x10E41DF4 (FORCE_DLAA) - boolean
0x10E41DF5 ("scaling ratio from DRS") - float
0x10AFB76A ("Fast UAV clear") - dword flags (1 / 2 / 4), might be existing setting that DLSS now taps into?
0x10AFB76B - string, logging related, might be log filename
0x10AFB76C (OVERRIDE_PERF_TO_9X) - boolean
```

FORCE_DLAA & "scaling ratio from DRS" both seem to work fine, but unfortunately only when setting them on the base profile, doesn't look like DLSS tries reading the profile for the current game at all ;_;

Would be great if these can be added, I'm not sure if profile inspector supports float values though, converting it from float to int first was needed.

**Comment by emoose** [comment]
Here's a CustomSettingNames.xml file which contains the DLAA/ratio/preset options, extract this next to "nvidiaProfileInspector.exe" and they should appear under "5 - Common" group.

[CustomSettingNames-DLSS.zip](https://github.com/Orbmu2k/nvidiaProfileInspector/files/12234465/CustomSettingNames-DLSS.zip)

Not sure if inspector can support floats, so I just added a few different ratios that I think people would use, 0.33 / 0.5 / 0.75 / 0.8 / 0.85 / 0.9 / 0.95 / 0.99 / 1.00.

Make sure you only apply this on the `_GLOBAL_DRIVER_PROFILE (Base Profile)` profile, since DLSS only seems to read from that and doesn't check the game specific one for some reason.

**Comment by OliverMD15** [comment]
I quickly tried it out in "Battlefield 2042", forcing DLAA seems to work there without any issues. 
I'm not entirely sure about "Forced Scaling" though, but at least it's a promising start.

I then tried verify my findings in "Baldur's Gate 3" using "DLSSTweaks" and the DLSS debug overlay. 
Forced Scaling functions smoothly, as does enforcing specific presets from what I can see. 

Now, if you could provide me with a 0.625 option, that would be all I'm missing personally.
For context: BF2042 used version 3.1.11 and BG3 version 3.1.30

**Comment by emoose** [comment]
> Now, if you could provide me with a 0.625 option, that would be all I'm missing personally. For context: BF2042 used version 3.1.11 and BG3 version 3.1.30

Sorry for being late with it, but should be able to use that by changing one of the HexValue settings in that XML to:

`<HexValue>0x3F200000</HexValue>`

**Comment by OliverMD15** [comment]
No reason to apologize. Before asking for help, I tried to fix it myself but always failed at the conversion for some reason or another. Anyway, now it's fixed and working – all that matters in the end!

**Comment by OliverMD15** [comment]
Interesting, I never thought about using DLSS to only upscale one axis. 
I usually just go with .625 and call it a day. Thanks for the idea!

**Comment by TheOriginalBabu** [comment]
Ultra Quality - 0.77x might also be useful to add, since it's a semi-official preset.

**Comment by RealSankiri** [comment]
Nice work :) When will be get Nvidia inspector Profile with above 3.5.xx?

**Comment by yamaci17** [comment]
Hello, emoose, thanks for this XML file. What I would like to ask is if it is possible to add extra options for DLSS ratios such as 0.25x and 0.4x. 0.25x for fun reasons and 0.4x can be a nice alternative to 0.33x

**Comment by Foul-Tarnished** [comment]
Still need to set force dlaa on base profile ?
Only nvidia can fix this ?

**Comment by Sh1lov8** [comment]
What does each of these presets mean? A B C D E(united) F G(unused)

**Comment by Foul-Tarnished** [comment]
Sir this is not google

**Comment by BlackSkillX** [comment]
Sadly you can not force a scaling ratio (for example 0.99x) in the Nvidia Inspector (with the XML extension) when using the new DLSS version 3.8.10 .
Not sure if it a hard-limitation of the new DLSS version or if the Nvidia Inspector just needs an update/newer XML version.
I would appreciate if scaling ratio could be made possible again with DLSS version 3.8.10.

**Comment by Foul-Tarnished** [comment]
As a workaround you can do this with either SpecialK or DLSS Tweaker

SpecialK work with most anti-cheats (like RTSS), but it's still a risk, as
it's hooking the game and injecting code

Le ven. 15 nov. 2024 à 20:51, BlackSkillX ***@***.***> a
écrit :

**Comment by BlackSkillX** [comment]
Both SpecialK or DLSS Tweaker do not work in the Finals. And some other games with easy anti cheat as well.

**Comment by paperclip-dayo** [comment]
Yup, seems like 0x10E41DF5 for forcing a scaling ratio is ignored by the DLSS 3.8.10, other flags like 0x10E41DF4 (force DLAA) and 0x10AFB76C (perf -> ultraperf) work fine.
DLSSTweaker also does not work, by that I mean the forcing scaling globally without the dll injection and not per-profile manual scaling that requires injection. Latter one still works but it's not feasible for anti-cheat protected games. 
I assume Nvidia just removed this "feature" whether intentionally or unintentionally when they were cleaning up available presets.

**Comment by paperclip-dayo** [comment]
Yup, seems like 0x10E41DF5 for forcing a scaling ratio is ignored by the DLSS 3.8.10, other flags like 0x10E41DF4 (force DLAA) and 0x10AFB76C (perf -> ultraperf) work fine.
DLSSTweaker also does not work, by that I mean the forcing scaling globally without the dll injection and not per-profile manual scaling that requires injection. Latter one still works but it's not feasible for anti-cheat protected games.
Hopefully @emoose can take a closer look at it when he gets some free time but I assume Nvidia just removed this "feature" whether intentionally or unintentionally when they were cleaning up available presets.

**Comment by xHybred** [comment]
Hey emoose, can I ask how you found the HexSettingID for these commands?

I'm trying to locate the hex for the new NVIDIA app DLSS overrides. I'd appreciate being made aware of your methodology

**Comment by Egor179** [comment]
What HEX code do I need to enter to get an input resolution of 1918x1078?
(100%/DLAA does not work, DLSS simply turns off)
99% is 1901x1069
I tried the IEEE 754 Converter
99.9 converted to 42C7CCCD, but the final resolution was 960x540
I used 2.4.0.29 NPI
